src/map/core.rs

```diff
@@ -754,9 +754,128 @@ impl<K: fmt::Debug, V> fmt::Debug for VacantEntry<'_, K, V> {
     }
 }
 
+/// A view into an occupied entry in a `IndexMap` obtained by index.
+///
+/// This struct is constructed from the `entry_at` method on `IndexMap`.
+// SAFETY: We have a mutable reference to the map, which keeps the index
+// valid and pointing to the correct entry.
```

Comment:
We don't need a "SAFETY" comment, because this is all safe code. In `OccupiedEntry`, which I assume you were copying, it's concerned with the validity of the `unsafe` access through its raw bucket.

If you still want a comment about the correctness of the index, I'd move that to the field and drop the "SAFETY" connotation.